emir-assignment/src/lib/tmdb.ts

```diff
@@ -0,0 +1,129 @@
+const BASE_URL = "https://api.themoviedb.org/3";
+const API_KEY = import.meta.env.VITE_TMDB_API_KEY as string | undefined;
+
+if (!API_KEY) {
+    console.error("[TMDB] Missing VITE_TMDB_API_KEY in .env");
+}
+
+function toSearchParams(params?: Record<string, unknown>) {
+    const sp = new URLSearchParams();
+    if (!params) return sp;
+
+    Object.entries(params).forEach(([k, v]) => {
+        if (v === undefined || v === null) return;
+        sp.set(k, String(v));
+    });
+    return sp;
+}
+
+type FetchJsonInit = Omit<RequestInit, "headers"> & {
+    signal?: AbortSignal;
+};
+
+/**
+ * Minimal fetch wrapper for TMDB v3.
+ * - Always appends api_key param
+ * - Adds default params: language=en-US, include_adult=false
+ * - Throws on non-2xx with useful error info
+ */
+export async function fetchJson<T = unknown>(
+    path: string,
+    params?: Record<string, unknown>,
+    init?: FetchJsonInit
+): Promise<T> {
+    const url = new URL(path.replace(/^\//, ""), BASE_URL);
+
+    const search = toSearchParams({
+        api_key: API_KEY,
+        language: "en-US",
+        include_adult: false,
+        ...params,
+    });
+
+    url.search = search.toString();
+
+    const res = await fetch(url.toString(), init);
```

Comment:
Nowadays I would want to use react-query and build the tmdb lib around userQuery primitives with hooks like `useMovies` and `useMovieDetails` or something like that, you get caching OOTB that way.